google/cloud/bigquery/job/query.py

```diff
@@ -1507,6 +1512,15 @@ def result(  # type: ignore  # (incompatible with supertype)
                 # Intentionally omit job_id and query_id since this doesn't
                 # actually correspond to a finished query job.
             )
+
+        # When timeout has default sentinel value, only pass the sentinel
+        # timeout to QueryJob.done(), and use None for the other timeouts.
+        if type(timeout) is object:
+            timeout = None
+            get_job_timeout = PollingFuture._DEFAULT_VALUE
+        else:
+            get_job_timeout = timeout
```

Comment:
Why not do the kwargs trick here too?

Reply:
I wonder if it's necessary - the purpose here is slightly different from when calling `client.get_job()`. I'm only preserving the sentinel value for the `job.done()` call, and use None as default for any other calls.

Reply:
done, following later comments